src/Calculator.ts

```diff
@@ -268,35 +288,54 @@ export async function calculateAggregateMeasureReport(
     debugOutput.measureReports = [report];
   }
 
-  return { results: report, debugOutput };
+  return { results: report, debugOutput, valueSetCache: calculationResults.valueSetCache };
 }
 
+/**
+ * Get raw results from cql-execution
+ *
+ * @param measureBundle Bundle with a MeasureResource and all necessary data for execution.
+ * @param patientBundles List of bundles of patients to be executed.
+ * @param options Options for calculation.
+ * @param valueSetCache Cache of existing valuesets
+ * @returns pass through of raw calculation results from the engine
+ */
 export async function calculateRaw(
   measureBundle: R4.IBundle,
   patientBundles: R4.IBundle[],
-  options: CalculationOptions
-): Promise<{ results: cql.Results | string; debugOutput?: DebugOutput }> {
+  options: CalculationOptions,
+  valueSetCache: R4.IValueSet[] = []
+): Promise<{ results: cql.Results | string; debugOutput?: DebugOutput; valueSetCache?: R4.IValueSet[] }> {
```

Comment:
Still trying to learn the code, could you explain what this does?

Reply:
This function is the main entry point for simply running the measure logic against a list of patients and returning the raw results from the `cql-execution` engine. These raw results don't have any processing on them to relate the results to the populations defined in the measure.

Pretty much all of the work that this function returns is in done in the `execute` function in `Execution.ts`. This function just formats it correctly. In this PR it passes back the valueSetCache if one was created and filled with any valuesets pulled from VSAC.